tested/languages/typescript/templates/values.ts

```diff
@@ -0,0 +1,146 @@
+        value = value.toString();
+    } else if (typeof value === "symbol") {
+        type = "unknown";
+        value = value.toString();
+    } else if (typeof value === "object") {
+        if (value === null) {
+            type = "null";
+        } else if (Array.isArray(value)) {
+            type = "list";
+            // Handle holes in arrays...
+            const unholed = [];
+            for (let i = 0; i < value.length; i++) {
+                if (!(i in value)) {
+                    unholed.push(`<empty at index ${i}>`)
+                } else {
+                    unholed.push(value[i]);
+                }
+            }
+            value = unholed.map(encode);
+        } else if (value instanceof Set) {
+            type = "set";
+            value = Array.from(value).map(encode);
+        } else if (value instanceof Map) {
+            type = "dictionary";
+            value = Array
+                    .from(value)
+                    .map(([key, value]) => {
+                                return {
+                                    key: encode(key),
+                                    value: encode(value)
+                                };
+                            }
+                    );
+        } else if (value?.constructor === Object || isVanillaObject(value)) {
+            // Plain objects
+            type = "object";
+            // Process the elements of the object.
+
+            value = Object.keys(value).map(key => {
+                return {
+                    key: encode(key),
+                    value: encode((value as Record<string, any>)[key])
+                };
+            });
+        } else {
+            type = "unknown";
+            diagnostic = value?.constructor?.name;
+            value = JSON.stringify(value);
+        }
+    } else {
+        type = "unknown";
+        diagnostic = (value as Object)?.constructor?.name;
+        value = Object.prototype.toString.call(value);
+    }
+
+    return {
+        type: type,
+        data: value,
+        diagnostic: diagnostic
+    };
+
+}
+
+// Send a value to the given stream.
+function sendValue(stream: number, value: Object) {
+    fs.writeSync(stream, JSON.stringify(encode(value)));
+}
+
+// Send an exception to the given stream.
+function sendException(stream: number, exception: Error | Object | {constructor: {name: any}}): void {
+    if (!exception) {
+        return;
+    }
+    if (exception instanceof Error) {
+        // We have a proper error...
+        fs.writeSync(stream, JSON.stringify({
+            "message": exception.message,
+            "stacktrace": exception.stack ?? "",
+            "type": exception.constructor.name
+        }));
+    } else {
+        // Comes out of the values.js:
+        // Temporarily allow objects with "message" and "name".
+        // TODO: remove this once the semester is over
+        // noinspection PointlessBooleanExpressionJS
+        if (typeof exception === 'object') {
```

Comment:
I would remove this special case: it was added to not change the behaviour of JavaScript judging in the middle of the course, so it can probably be removed from JavaScript (and no need to add it here to TypeScript)

Reply:
What should I do about those tests that depend on them?

Reply:
I would update the submissions that are being tests to account for the new behaviour (the PR that introduced this is https://github.com/dodona-edu/universal-judge/pull/361, which might be helpful): all exceptions must inherit from `Error`.

It is possible that some example solutions in the js repo will need to be updated as well. In that case, I would not remove this for JS in this PR, but create an issue for this and remove the special case in JS in another PR (this one is big enough as it is)